src/engine/guidance/post_processing.cpp

```diff
@@ -497,11 +499,43 @@ void collapseTurnAt(std::vector<RouteStep> &steps,
     }
 }
 
+// Works on steps including silent and invalid instructions in order to do lane anticipation for
+// roundabouts which later on get collapsed into a single multi-hop instruction.
+std::vector<RouteStep> anticipateLaneChangeForRoundabouts(std::vector<RouteStep> steps)
+{
+    util::guidance::print(steps);
+
+    const constexpr auto MIN_DURATION_NEEDED_FOR_LANE_CHANGE_INSIDE_ROUNDABOUT = 15.;
```

Comment:
within a roundabout, I would account deem all lane changes relevant, since you are in a continuous turn.